### PR TITLE
Making Stagger Great Again

### PR DESCRIPTION
**Pull request by Surrealaser**
## About The Pull Request

Stagger is now meaningful for carbons per the following:

Halves all projectile damage after all other mods due to the concussion turning hits into glancing ones.

## Why It's Good For The Game

Makes Stagger meaningful as a debuff for non-Xeno targets. Also opens up debuff based counterplay to point blank builds.

Makes Stagger a notable substitute for debuffs/control CC in lieu of stuns.

## Changelog
:cl:
tweak: The Stagger debuff now halves all projectile damage.
/:cl:

### CODE REVIEW

File: code/modules/projectiles/projectile.dm [hunk 1]
```diff
@@ -663,14 +663,17 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
...
-			if(shooter_human.stagger)
-				BULLET_DEBUG("Stagged (-30).")
-				. -= 30 //Being staggered fucks your aim.
 			if(shooter_human.marksman_aura) //Accuracy bonus from active focus order: flat bonus + bonus per tile traveled
 				BULLET_DEBUG("marksman_aura (+[shooter_human.marksman_aura * 3] + [proj.distance_travelled * shooter_human.marksman_aura * 0.35]).")
 				. += shooter_human.marksman_aura * 3
 				. += proj.distance_travelled * shooter_human.marksman_aura * 0.35
 
+		if(iscarbon(proj.firer))
+			var/mob/living/carbon/human/shooter_carbon = shooter_living
```
Comment (Hughgent) [inline code/modules/projectiles/projectile.dm:672]: You've moved this, checked if its carbon, then typecast as human.

what's up with all that?

Comment (Surrealaser) [reply code/modules/projectiles/projectile.dm:672]: wtf

Comment (Surrealaser) [reply code/modules/projectiles/projectile.dm]: Die copy pasta scum

**Review by Hughgent — APPROVED** [review]
I'm ok with this.

**Comment by SplinterGP** [comment]
should add the PB change to changelog too.

**Comment by SplinterGP** [comment]
oh wait nvm misread.

**Review by Hughgent — APPROVED** [review]

**Review by TiviPlus — CHANGES_REQUESTED** [review]
Atomize

**Comment by Surrealaser** [comment]
> Atomize

Both components are intended to constitute the overall effect of stagger; the reduction of accuracy and denial of auto-hits on point blank shots.

**Comment by spookydonut** [comment]
> 
> 
> Atomize

stop being a nerd.

**Comment by TiviPlus** [comment]
PB is supposed to be fully accurate and I dont want to yell on this PR about it

**Comment by Surrealaser** [comment]
> PB is supposed to be fully accurate and I dont want to yell on this PR about it

There, now it halves projectile damage instead of incrementing RNG.

**Review by TiviPlus — APPROVED** [review]

File: code/modules/projectiles/projectile.dm [hunk 2]
```diff
@@ -792,6 +789,11 @@ So if we are on the 32th absolute pixel coordinate we are on tile 1, but if we a
 			if(living_soft_armor)
 				living_soft_armor = max(0, living_soft_armor - penetration) //Flat removal.
 
+		var/mob/living/carbon/shooter_carbon = proj.firer
+		if(iscarbon(shooter_carbon))
```
Comment (Hughgent) [inline code/modules/projectiles/projectile.dm:793]: typecasting before checking could lead to runtimes if I recall correctly.

**Review by Hughgent — CHANGES_REQUESTED** [review]

File: code/modules/projectiles/projectile.dm (hunk 2 above)
Comment (Surrealaser) [reply code/modules/projectiles/projectile.dm:793]: Should be fine; it would get caught in the check, but I don't mind changing this.

**Review by Hughgent — APPROVED** [review]

**Review by TiviPlus — APPROVED** [review]

**Pull request merged**